src/unix/uiactionx11.cpp

```diff
@@ -46,6 +55,113 @@ GtkWidget* wxGetTopLevelGTK();
 
 namespace
 {
+// Like the real events, this class tries to add a _fake_ delay to the generated
+// (fake) events so that chances that they got lost or ignored (for whatever reasons)
+// by the X server (or the WM, or even the input device driver) are minimized.
+class wxXSync
+{
+public:
+    wxXSync(wxX11Display& display)
+        : m_display(display), m_isMotion(true)
+    {
+    }
+
+    wxXSync(wxX11Display& display, bool depressed)
+        : m_display(display), m_isMotion(false)
+    {
+        static int s_maxDepressed = 0;
+
+        if ( depressed )
+        {
+            ++ms_numDepressed; // A key or button is down.
+
+            if ( ms_numDepressed - s_maxDepressed != 1 )
+                wxFAIL_MSG("Invalid call to wxXSync() ctor");
+
+            s_maxDepressed = ms_numDepressed;
+        }
+        else
+        {
+            if ( s_maxDepressed != 0 )
+            {
+                s_maxDepressed = 0; // reset
+
+                wxYield();
+                wxMilliSleep(Default_Delay);
+            }
+
+            --ms_numDepressed; // A key or button is up.
+
+            if ( ms_numDepressed < 0 )
+                wxFAIL_MSG("Invalid call to wxXSync() ctor");
+        }
+    }
+
+    ~wxXSync()
+    {
+        XSync(m_display, False);
+
+        m_isMotion ? OnMotion() : OnKeyOrButtonUp();
+    }
+
+private:
+    static void OnMotion()
+    {
+        wxMilliSleep(Motion_Delay);
+        wxYield();
```

Comment:
It's interesting that for the motion events you sleep first and then yield, while for the other ones it's the reverse. Do you have any idea about why does it need to be done like this? If so, it would be great to have a comment explaining it here.

Reply:
My fault, this is unintentionally reversed.
but now I am wondering what would happen if we just `wxYield`ing without any delay at all ?
I'll take a closer look.

Reply:
I removed the sleep entirely now without apparent consequences. seems not needed anymore.